core/src/main/java/com/klaytn/caver/tx/type/AbstractTxType.java

```diff
@@ -84,14 +85,59 @@ public AbstractTxType(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit
         this.from = from;
         this.to = to;
         this.value = value;
+        this.senderSignatureDataList = new ArrayList();
     }
 
     protected void setSenderSignatureData(KlaySignatureData signatureData) {
-        this.senderSignatureData = signatureData;
+        addSenderSignatureData(signatureData);
     }
 
     public KlaySignatureData getSenderSignatureData() {
-        return senderSignatureData;
+        try {
+            return senderSignatureDataList.get(0);
+        } catch (Exception e){
+            return null;
+        }
+    }
+
+    protected void addSenderSignatureData(KlaySignatureData signatureData) {
+        for (KlaySignatureData senderSignatureData : senderSignatureDataList) {
+            if (senderSignatureData.equals(signatureData)) {
+                return;
+            }
+        }
+        senderSignatureDataList.add(signatureData);
+    }
+
+    public void addSenderSignatureDataList(List<KlaySignatureData> signatureDataList) {
+        for (KlaySignatureData signatureData : signatureDataList) {
+            addSenderSignatureData(signatureData);
```

Comment:
Don't we need to check duplication?

Reply:
Oh, that could be done by addSenderSignatureData(). Please ignore my comment. :)